person.rb

```diff
@@ -18,6 +25,31 @@ def can_use_services?
   end
 
   private :of_age?
+
+  def correct_name
+    @name
+  end
 end
 
-Person.new(14, 'Jonathan')
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.chars.slice(0, 9).join
+  end
```

Comment:
![image](https://user-images.githubusercontent.com/93039359/216044453-cadddfbc-c39e-47e2-bbfa-f5cb4915fa72.png)


- [ ] Great work so far on implementing this method but i guess missed that your trimmer should trim after 10 characters because this is returning only 9 characters from start, not 10. Kindly fix it by adding 10 instead of 9 in the method above :+1: